File: ruby/lib/urlopener.rb
# Created by Satoshi Nakagawa.
# You can redistribute it and/or modify it under the Ruby's license or the GPL2.

require 'preferences'

class UrlOpener < NSObject
  def self.openUrl(str)
    if preferences.general.open_browser_in_background
      urls = [OSX::NSURL.URLWithString(str)]
      OSX::NSWorkspace.sharedWorkspace.openURLs_withAppBundleIdentifier_options_additionalEventParamDescriptor_launchIdentifiers(urls, nil, NSWorkspaceLaunchWithoutActivation, nil, nil)
    else
      url = OSX::NSURL.URLWithString(str)
      OSX::NSWorkspace.sharedWorkspace.openURL(url)
    end
  end
end

File: Classes/Helpers/URLOpener.m
// LimeChat is copyrighted free software by Satoshi Nakagawa <psychs AT limechat DOT net>.
// You can redistribute it and/or modify it under the terms of the GPL version 2 (see the file GPL.txt).

#import "URLOpener.h"
#import "Preferences.h"


@implementation URLOpener

+ (void)open:(NSURL*)url
{
	NSWorkspace* ws = [NSWorkspace sharedWorkspace];
	
	if ([Preferences openBrowserInBackground]) {
		[ws openURLs:[NSArray arrayWithObject:url] withAppBundleIdentifier:nil options:NSWorkspaceLaunchWithoutActivation additionalEventParamDescriptor:nil launchIdentifiers:nil];
	}
	else {
		[ws openURL:url];
	}
}

+ (void)openAndActivate:(NSURL*)url
{
	[[NSWorkspace sharedWorkspace] openURL:url];
}

@end
